.github/workflows/integration_tests.yml

```diff
@@ -37,9 +37,17 @@ jobs:
     runs-on: ${{ needs.start-runner.outputs.label }}
     timeout-minutes: 600
     steps:
+      - name: Determine branch ref
+        id: branch_ref
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "branch_ref=refs/heads/${{ github.head_ref }}" >> $GITHUB_OUTPUT
```

Comment:
We had to do some gymnastics here because, for a pull request, `github.ref` evaluates to `refs/pull/<pr_number>/merge`. While `github.head_ref` evaluates to the short name for the branch (just `<name of the branch>`). Since we require the fully formed `refs/heads/<name of the branch>`, and no github variable can directly get us this, we do some string formatting and save the output to `$GITHUB_OUTPUT`.